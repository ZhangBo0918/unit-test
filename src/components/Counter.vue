<template>
  <div>
    <span>Count:{{ count }}</span>
    <button @click="count++">按钮</button>
  </div>
</template>

<script>
export default {
  data() {
    return {
      count: 0,
    };
  },
};
</script>

<style scoped>
</style>