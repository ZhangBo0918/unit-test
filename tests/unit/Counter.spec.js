/**
 * @jest-environment jsdom
 */
import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter.vue", () => {
  const wrapper = mount(Counter);
  it("renders props.count when passed", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("count++", () => {
    const button = wrapper.find('button');
    button.trigger("click");
    expect(wrapper.vm.count).toBe(1);
  })
});
