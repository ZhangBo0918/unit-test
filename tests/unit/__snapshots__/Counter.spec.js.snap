// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders props.count when passed 1`] = `"<div><span>Count:0</span> <button>按钮</button></div>"`;
